scripts/restore

```diff
@@ -17,14 +17,133 @@ source /usr/share/yunohost/helpers
+ynh_print_info "Restoring the app main directory..."
+
+ynh_restore_file "$final_path"
+
+if [[ $firmware_nonfree -eq 1 ]]; then
+  # check if non-free is set on sources.list
+  if ! grep -q non-free /etc/apt/sources.list ; then
+    sed '/debian/{s/main/& non-free/}' -i /etc/apt/sources.list
+  fi
+
+  packages=$nonfree_packages
+else
+  packages=$free_packages
+
+  ynh_restore_file "/lib/firmware/htc_7010.fw"
+  ynh_restore_file "/lib/firmware/htc_9271.fw"
+fi
+
+ynh_restore_file "/etc/sudoers.d/${app}_ynh"
+
+ynh_restore_file "/usr/local/bin/iw_multissid"
+ynh_restore_file "/usr/local/bin/iw_devices"
+ynh_restore_file "/usr/local/bin/iw_ssids"
+ynh_restore_file "/usr/local/bin/ipv6_expanded"
+ynh_restore_file "/usr/local/bin/ipv6_compressed"
+
+for FILE in $(ls /etc/hostapd/hostapd.conf{.tpl?,} 2>/dev/null)
+do
+    ynh_restore_file "$FILE"
+done
+ynh_restore_file "/etc/dnsmasq.dhcpd/dhcpdv6.conf.tpl"
+ynh_restore_file "/etc/dnsmasq.dhcpd/dhcpdv4.conf.tpl"
+
+ynh_restore_file  "/usr/local/bin/$service_name"
+
+ynh_restore_file "/etc/init.d/hostapd"
+
+#=================================================
+# RECREATE THE DEDICATED USER
+#=================================================
+ynh_print_info "Recreating the dedicated system user..."
 
-gitcommit=$(sudo grep revision /etc/yunohost/apps/hotspot/status.json | sed 's/.*"revision": "\([^"]\+\)".*/\1/')
-tmpdir=$(mktemp -dp /tmp/ hotspot-restore-XXXXX)
+# Create the dedicated user (if not existing)
+ynh_system_user_create $app
 
-git clone https://github.com/labriqueinternet/hotspot_ynh.git "${tmpdir}/"
-git --work-tree "${tmpdir}/" --git-dir "${tmpdir}/.git/" reset --hard "${gitcommit}"
+#=================================================
+# RESTORE USER RIGHTS
+#=================================================
+
+# Restore permissions on app files
+chown -R www-data: $final_path
+
+#=================================================
+# RESTORE THE PHP-FPM CONFIGURATION
+#=================================================
+
+ynh_restore_file "/etc/php/7.0/fpm/pool.d/$app.conf"
 
-cd "${tmpdir}/scripts/"
-bash ./upgrade
+#=================================================
+# SPECIFIC RESTORATION
+#=================================================
+# REINSTALL DEPENDENCIES
+#=================================================
+ynh_print_info "Reinstalling dependencies..."
+
+# Define and install dependencies
+ynh_install_app_dependencies "$pkg_dependencies" "$packages"
+
+#=================================================
+# RESTORE SYSTEMD
+#=================================================
+ynh_print_info "Restoring the systemd configuration..."
+
+ynh_restore_file "/etc/systemd/system/$service_name.service"
+systemctl enable $service_name.service
+
+#=================================================
+# ADVERTISE SERVICE IN ADMIN PANEL
+#=================================================
+
+yunohost service add $app --log "/var/log/$app/$app.log"
```

Comment:
We absolutely need the `--need_lock`  here as in the install :stuck_out_tongue: (also would be nice to add the description as well)